## Code samples

The above is frontmatter that might be needed by devportal when exported; ignore it for now. The rest of this notebook would be visible to the customer, as is. First, let's see Python work.

In [1]:
print("hello world")

hello world


For me it does; it says hello world below the python cell. Yay! One could change hello world to any other string to see dynamic running code, in its most fundamental form.
## Splunk Observability Cloud

The code below sends metric data to your Splunk Obvervability Cloud instance. Within 'https://ingest.us1.signalfx.com', replace 
_us1_ with your realm, _j195EsqEpXQgHQhec9P2ZA_ with your access token, and _you_ with your alias. If you don't do this...it actually sends metric data to my instance so thanks for that

In [31]:
import time
import signalfx

st = time.time()

sum_x = 0

for i in range(1000000):
    sum_x += i

# wait for 3 seconds
time.sleep(3)

print('Sum of first 1 million numbers is:', sum_x)

et = time.time()

elapsed = et - st

with signalfx.SignalFx(ingest_endpoint='https://ingest.us1.signalfx.com').ingest('j195EsqEpXQgHQhec9P2ZA') as sfx:
    sfx.send(
        gauges=[
          {
                "metric": "execution.time",
                "dimensions": { "host": "you" },
                "value": et - st
          }
        ])

print("Time sent to Splunk O11y")

Sum of first 1 million numbers is: 499999500000
Time sent to Splunk O11y


## Splunk Python SDK

The following code processes data from a Splunk Enterprise search using the Splunk Python SDK.

If you receive something like "ModuleNotFoundError: No module named 'pandas'" that means you must first install the missing library. Meaning, from your command prompt enter ``pip install pandas``, ``pip install splunk-sdk`` etc. If you don't even have pip, you're likely not a Python coder yet...you need to first run ``python get-pip.py``. If you don't have Python yet, you're definitely not a Python coder yet so go install it but you're probably not the audience for python samples in a python-based notebook...And as it will tell you: you may need to restart the Notebook kernel to use updated packages (__Kernel__, __Restart Kernel__).

In [2]:
import pandas as pd
import numpy as np
import splunklib.client as client
import splunklib.results as results
import requests
from time import sleep

Now its time to connect to your Splunk Enterprise instance! In the following code, replace pking and Nighthawk700 with your own Splunk Enterprise login credentials

In [23]:
HOST = "localhost"
PORT = 8089
USER_NAME = "pking"
PASSWORD = "Nighthawk700"
service = client.connect(host=HOST, port=PORT, username=USER_NAME, password=PASSWORD)

Now, let's do something real in this notebook from Splunk Enterprise.

In [30]:
job = service.jobs.create('search index=_internal ( log_level=ERROR OR log_level=WARN ) | stats count')
while not job.is_done():
    sleep(.2)
rr = results.JSONResultsReader(job.results(output_mode='json'))
matches = pd.DataFrame()
for result in rr:
    print(result)
assert rr.is_preview == False

{'count': '360921'}


There you go; the count of existing warnings in your internal Splunk index! For me, when I run it, the count was 360921. Next time it should differ. But...Data! That's the point.